### ISSUE TITLE
Excluding files from analysis also excludes them from analysis classpath

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt#5555]
## Expected Behavior
When I exclude a file from analysis by doing something like

```kotlin
tasks.detektMain {
    exclude("**/BaseRule.kt")
}
```
Then `BaseRule.kt` should be excluded from analysis, but should still be on the analysis classpath.

## Observed Behavior
Excluding the file means analysis on files that call `BaseRule` do not have issues flagged, when they should.

## Steps to Reproduce
Using the detekt project itself:
1. Add `io.gitlab.arturbosch.detekt.api.BaseRule.visitFile` to `ForbiddenMethodCall` methods list in `detekt.yml`
2. Add this code to detekt-api's `build.gradle.kts`:
```
tasks.detektMain {
    exclude("**/BaseRule.kt")
}
```
3. Run `detektMain`. No issues are reported. This is incorrect.

What should happen at step 3 is that the task fails with error `C:\path\to\detekt\detekt-api\src\main\kotlin\io\gitlab\arturbosch\detekt\api\RuleSet.kt:36:20: The method 'io.gitlab.arturbosch.detekt.api.BaseRule.visitFile' has been forbidden in the detekt config. [ForbiddenMethodCall]`.

This error should still be shown because while `BaseRule` has been excluded from analysis, `RuleSet` still calls the forbidden `BaseRule.visitFile` method and should flag ForbiddenMethodCall.

## Context
In general this is likely low impact as I don't think many rules are directly affected, and they will present as false negatives not false positives. But this does highlight a bigger issue with the correctness of the classpath that's used by detekt when it generates BindingContext for rules that use type resolution.

## Your Environment

* Version of detekt used: 1.22.0
* Version of Gradle used (if applicable): 7.5.1
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): N/A
